test/linalg/special.jl

```diff
@@ -1,59 +1,54 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 using Base.Test
-debug = false
 
 n= 10 #Size of matrix to test
 srand(1)
 
-debug && println("Test interconversion between special matrix types")
-let a=[1.0:n;]
-   A=Diagonal(a)
-   for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
-       debug && println("newtype is $(newtype)")
+@testset "Interconversion between special matrix types" begin
+   a = [1.0:n;]
+   A = Diagonal(a)
+   @testset for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
```

Comment:
I feel like we can live without the inner testsets in the `"Interconversion between special matrix types"` testset. Here for instance, we make 5 testsets with 1 test inside. There is also quite an overhead for each testset, so would be nice to limit the number of sets. As I said above, I think we can remove the inner testsets here, or if you like it better, simplify to something like:

```julia
@testset "converstion from Diagonal" begin
    for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
        @test full(convert(newtype, A)) == full(A)
    end
end
```
such that at least all of these tests gather in the same set.